File: bench/Reference/list_map_filter.erl
#!/usr/bin/env escript
%% List map (*2) then filter (divisible by 4), sum.

main(_) ->
    L = build(10000, []),
    Doubled = [X * 2 || X <- L],
    Evens = [X || X <- Doubled, X rem 4 =:= 0],
    io:format("~p~n", [lists:sum(Evens)]).

build(0, Acc) -> Acc;
build(N, Acc) -> build(N - 1, [N | Acc]).

File: bench/Reference/list_map_filter.hs
-- list_map_filter: map (*2) + filter (`mod` 4 == 0) on 1..10000, sum.
main :: IO ()
main = print (sum [x * 2 | x <- go 10000 [], (x * 2) `mod` 4 == 0])
  where
    go 0 acc = acc
    go n acc = go (n - 1) (n : acc)
